openlibrary/core/bookshelves.py

```diff
@@ -404,25 +477,32 @@ def get_sorted_reading_log_books(
                 oldb.query(query, vars=query_params)
             )
 
-            reading_log_work_keys = [
-                '/works/OL%sW' % i['work_id'] for i in reading_log_books
+            reading_log_keys = [
+                ('/works/OL%sW' % i['work_id'], '/books/OL%sM' % i['edition_id'])
+                for i in reading_log_books
             ]
+
             solr_docs = get_solr().get_many(
-                reading_log_work_keys,
+                [key for key_pair in reading_log_keys for key in key_pair],
                 fields=WorkSearchScheme.default_fetched_fields
                 | {'subject', 'person', 'place', 'time', 'edition_key'},
             )
             solr_docs = add_storage_items_for_redirects(
-                reading_log_work_keys, solr_docs
-            )
-            assert len(solr_docs) == len(reading_log_work_keys), (
-                "solr_docs is missing an item/items from reading_log_work_keys; "
-                "see add_storage_items_for_redirects()"
+                [keys[0] for keys in reading_log_keys if keys[0]], solr_docs
             )
 
             total_results = shelf_totals.get(bookshelf_id, 0)
             solr_docs = add_reading_log_data(reading_log_books, solr_docs)
 
+            # Attaches returned editions to works.
+            if editions:
+                solr_docs = cls.link_editions_to_works(solr_docs)
+
+            assert len(solr_docs) == len(reading_log_keys), (
```

Comment:
Something about this branch has caused the My Books page to error with at least 2 of my accounts.
![image](https://github.com/internetarchive/openlibrary/assets/28732543/a32986d1-502b-4d3a-a971-73bf4fdad758)

Reply:
I'll get to this immediately; I must have overlooked something regarding redirects. If you could do me a favor, are you able to send me the json contents of the reading logs of the accounts in question? Specifically, I'm interested in the work/edition keys involved.

Reply:
I sent it to you via Slack.  Correction: this was only happening for one account.

Reply:
@jimchamp Sorry to add another request, but I'm having trouble reproducing this.  If you could do me a brief favor, could you try this with `"solr_docs found: %s, reading_log_keys:%s" % (len(solr_docs), len(reading_log_keys))` added to the feedback for the assertion error?

Reply:
<img width="421" alt="image" src="https://github.com/internetarchive/openlibrary/assets/131627264/feb1361c-aae8-479a-b7f3-056a32a32986">
Found some undesired behavior right here, but I'm not sure it's the same bug, or the root of the same bug.

Reply:
Looks like there are 12 Solr docs and 6 reading log key objects.